relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -0,0 +1,441 @@
+//! Functionality for calculating if a trace should be processed or dropped.
+//!
+use std::convert::TryInto;
+
+use rand::{distributions::Uniform, Rng};
+use rand_pcg::Pcg32;
+use serde::{Deserialize, Serialize};
+
+use relay_common::{ProjectId, ProjectKey, Uuid};
+use relay_filter::GlobPatterns;
+
+/// A sampling rule defined by user in Organization options
```

Comment:
_nit_: Please add punctuation. Same for other doc comments.

```suggestion
/// A sampling rule defined by user in Organization options.
```